src/general/world_effects/patch_events/GlobalGlaciationEvent.cs

```diff
@@ -0,0 +1,284 @@
+    {
+        patch.CurrentSnapshot.Background = Background;
+    }
+
+    private void AdjustEnvironment(Patch patch)
+    {
+        bool hasTemperature =
+            patch.Biome.ChangeableCompounds.TryGetValue(Compound.Temperature, out var currentTemperature);
+        bool hasSunlight = patch.Biome.ChangeableCompounds.TryGetValue(Compound.Sunlight, out var currentSunlight);
+
+        if (!hasTemperature)
+        {
+            GD.PrintErr("Glaciation event encountered patch with unexpectedly no temperature");
+            return;
+        }
+
+        if (!hasSunlight)
+        {
+            GD.PrintErr("Glaciation event encountered patch with unexpectedly no sunlight");
+            return;
+        }
+
+        currentTemperature.Ambient = random.Next(0, 5);
+        currentSunlight.Ambient = 0.5f;
+
+        patch.Biome.ModifyLongTermCondition(Compound.Temperature, currentTemperature);
+        patch.Biome.ModifyLongTermCondition(Compound.Sunlight, currentSunlight);
+    }
+
+    /// <summary>
+    ///   Gets chunks from Iceshelf patch template and applies them to the patches.
+    /// </summary>
+    private void AddIceChunks(Patch patch)
+    {
+        var templateBiome = SimulationParameters.Instance.GetBiome(TemplateBiomeForIceChunks);
+        foreach (var configuration in IceChunksConfigurations)
+        {
+            var iceChunkConfiguration = templateBiome.Conditions.Chunks[configuration];
+            iceChunkConfiguration.Density *= 10;
+            patch.Biome.Chunks.Add(Prefix + configuration, iceChunkConfiguration);
+        }
+    }
+
+    private void LogEvent(Patch patch, double totalTimePassed)
+    {
+        patch.LogEvent(new LocalizedString("GLOBAL_GLACIATION_EVENT"),
+            true, true, "GlobalGlaciationEvent.svg");
+
+        if (patch.Visibility == MapElementVisibility.Shown)
+        {
+            targetWorld.LogEvent(new LocalizedString("GLOBAL_GLACIATION_EVENT_LOG", patch.Name),
+                true, false, "GlobalGlaciationEvent.svg");
+        }
+
+        patch.AddPatchEventRecord(WorldEffectVisuals.GlobalGlaciation, totalTimePassed);
+    }
+
+    private void LogHeadUpEventWarning()
+    {
+        targetWorld.LogEvent(new LocalizedString("GLOBAL_GLACIATION_EVENT_WARNING_LOG", generationsToTrigger),
+            true, true, "GlobalGlaciationEvent.svg");
+    }
+
+    private void LogBeginningOfGlaciation()
+    {
+        targetWorld.LogEvent(new LocalizedString("GLOBAL_GLACIATION_START_EVENT_LOG"),
+            true, true, "GlobalGlaciationEvent.svg");
+    }
+
+    private void LogEndOfGlaciation()
+    {
+        targetWorld.LogEvent(new LocalizedString("GLOBAL_GLACIATION_END_EVENT_LOG"),
+            true, true, "GlobalGlaciationEvent.svg");
+    }
+
+    private void FinishEvent()
+    {
+        hasEventAlreadyHappened = true;
+        foreach (var index in modifiedPatchesIds)
+        {
+            if (!targetWorld.Map.Patches.TryGetValue(index, out var patch))
```

Comment:
Actually related to my earlier comment, it kind of looks like you are using the IDs? But call them indexes??